tests/test_validators.py

```diff
@@ -76,8 +76,8 @@ def test_doesnt_pollute_model(self):
 # -----------------------------------
 
 class UniquenessTogetherModel(models.Model):
-    race_name = models.CharField(max_length=100)
-    position = models.IntegerField()
+    race_name = models.CharField(max_length=100, null=True)
+    position = models.IntegerField(null=True)
```

Comment:
Rather than reusing this, let's instead have a `NullUniquenessTogetherModel`.
It would be good to also add a field that isn't part of the uniqueness constraint.

    class NullUniquenessTogetherModel(models.Model):
        """
        Used to ensure that null values are not included when checking unique_together constraints.

        Ignoring items which have a null in any of the validated fields is the same behavior that database
        backends will use when they have the unique_together constraint added.

        Example case: a null position could indicate a non-finisher in the race, there could be many non-finishers
        in a race, but all non-NULL values *should* be unique against the given `race_name`.
        """
        date_of_birth = models.DateField(null=True)  # Not part of the uniqueness constraint
        race_name = models.CharField(max_length=100)
        position = models.IntegerField(null=True)

        class Meta:
             unique_together = ('race_name', 'position')

...

    def test_ignore_validation_for_null_fields(self):
        # None values that are on fields which are part of the uniqueness constraint
        # cause the instance to ignore uniqueness validation.
        UniquenessTogetherModel.objects.create(
            date_of_birth=datetime.date(2000, 1, 1)
            race_name='Paris Marathon',
            position=None
        )
        data = {'date': datetime.date(2000, 1, 1), 'race_name': 'Paris Marathon', 'position': None}
        serializer = UniquenessTogetherSerializer(data=data)
        assert serializer.is_valid()

    def test_do_not_ignore_validation_for_null_fields(self):
        # None values that are not on fields part of the uniqueness constraint do not
        # cause the instance to skip validation.
        UniquenessTogetherModel.objects.create(
            date_of_birth=datetime.date(2000, 1, 1)
            race_name='Paris Marathon',
            position=1
        )
        data = {'date': None, 'race_name': 'Paris Marathon', 'position': 1}
        serializer = UniquenessTogetherSerializer(data=data)
        assert not serializer.is_valid()